packages/argo-run/src/utilities.ts

```diff
@@ -57,6 +57,7 @@ export function shouldUseReact(): boolean | 'mini' {
 
 export interface CheckoutExtensionConfig {
   readonly extensionPoints: string[];
+  readonly metafields?: any;
```

Comment:
That's a good point. I don't recall us discussing about separating the metafields based on the resource type. On the backend, the lookup query is not as bad because we have DB indexes to help with the filtering. In terms of the config file tho, I'm thinking grouping them might be more convenient if they have metafields that span multiple resources rather than having to specify the namespace and key pairs for each resource type. 

I was curious about how many metafield namespace and key pairs span multiple resources 😆 so I did this [query](https://app.mode.com/shopify/reports/9d6e967b4722) and found that there are multiple metafields that fit this condition. Most of the namespace/key pairs span Product and ProductVariants.